cmd/destroy/destroy.go

```diff
@@ -193,7 +202,18 @@ func Destroy(ctx context.Context) *cobra.Command {
 				return err
 			}
 
-			return destroyer.destroy(ctx, options)
+			err = destroyer.destroy(ctx, options)
+
+			metadata := &analytics.DestroyMetadata{
+				Success: err == nil,
+			}
+			if _, ok := destroyer.(*localDestroyAllCommand); ok {
```

Comment:
🤔 could be add some method to `destroyer` so it returns the `bool` calling `destroyer.IsDestroyAll()` for example, `destroyer.IsRemote()`
 so we remove the if/elses

Reply:
I thought it in the beginning but opted to do it this way for several reasons:
- keep interface as small as possible. If we add a new destroyer we'd have to implement an empty function if we don't want to be tracked
- [Interfaces should have no knowledge of satisfying types](https://blog.boot.dev/golang/golang-interfaces/#2-interfaces-should-have-no-knowledge-of-satisfying-types). The next time we implement a new destroyer we'd have to implement these two new methods even if both of them are false
- Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle and just have it on one place 

I though about implementing the way you described since it was very descriptive but I think this approach is more maintainable in the long term. WDYT?

Reply:
makes sense :)